### ISSUE TITLE
FlashList not optimizing virtualization when nested

### ISSUE THREAD

**Issue by pfranklin-detechtion-com** [Shopify/flash-list#1432]
In react-native app, we are using an outer FlashList with a list of groups of items, displayed by List.Accordion (react-native-paper), and each group displays its list of items in another inner FlashList.

When the List.Accordion is expanded, the inner FlashList is not virtualizing the display of the items properly. It is immediately rendering and displaying the first batch of items, then calls renderItem() on all of the remaining items in the list all the way to the end. We have some groups with 1000's of items, so this takes 10+ seconds or more.

When using a FlashList not nested, just flat, the virtualization works smoothly. Logging the renderItem() function shows that it is just rendering each item as it scrolls closer to the viewable window. I created a snack that isolates this behavior: [https://snack.expo.dev/LWIV8yrONzZw9Hshxm7f4](url)

Is there some combination of properties on the FlashList or other components that can achieve the desired behavior here, so that the items are virtualized in a performant way?

Or is this just an inherent problem when nesting lists of variable length items? Is there a better way to approach this?

**Comment by Daydreamer-riri** [comment]
Same question. For now, I have to give up this part of virtualization. I'm guessing flashlist doesn't support this behavior at the moment?

**Comment by pfranklin-detechtion-com** [comment]
Yes, also abandoned trying to make this work. I ended up putting it all in a single flat FlashList and adding removing the items from that list as the List.Accordions are pressed. So the List.Accordions are just there for looks and I handle the clicks and add/remove items under them. This was more complex (obviously) because I had to manage that myself, but it ended up working well and was very performant even with 1000's of items.

**Comment by vanBlayt** [comment]
u have to limit inner list's height. do u?

**Comment by daveskybet** [comment]
I'm suffering from the same issue, which also extends to `onViewableItemsChanged` seeing every item as viewable straight away unless a height is added to the inner list (as @vanBlayt mentioned), which doesn't help if you need to have multiple nested lists rendering in a performant way without nested scrollbars (as i need also need to). 

Plain `FlatList` nested supports this well, but this issue puts a hard block on migrating to something more performant than `FlatList`

**Comment by naqvitalha** [comment]
Hi! 👋

We're now focusing exclusively on **FlashList v2** and **React Native's New Architecture**. This means we're closing older issues and suggestions that aren't relevant to v2.

**If this issue/suggestion is still relevant:**
- Please verify it still occurs in FlashList v2
- Open a new issue with v2-specific details
- We'll be happy to take a look!

**Resources:**
- [Migration to v2](https://shopify.github.io/flash-list/docs/migration-guide)
- [FlashList Documentation](https://shopify.github.io/flash-list/)

Thank you for your understanding! 🙏
